octoprint_SpoolManager/static/js/SpoolManager.js

```diff
@@ -398,28 +397,53 @@ $(function() {
             return remainingInfo;
         }
 
-        self.selectSpoolForSidebar = function(spoolItem){
+        self.getSpoolItemSelectedTool = function(databaseId) {
+            var spoolItem;
+            for (var i=0; i<self.selectedSpoolsForSidebar().length; i++) {
+                spoolItem = self.selectedSpoolsForSidebar()[i]();
+                if (spoolItem !== null && self.selectedSpoolsForSidebar()[i]().databaseId() === databaseId) {
+                    return i;
+                }
+            }
+            return null;
+        }
+
+        self.selectSpoolForSidebar = function(toolIndex, spoolItem){
+            var commitCurrentState;
+            if (self.printerStateViewModel.isPrinting()) {
+                commitCurrentState = confirm(
+                    'You are changing a spool while printing. SpoolManager will commit the usage so far to the previous spool, unless you wish otherwise.\n\n' +
+                    'Commit the usage of the print so far…\n' +
+                    '"OK": …to the previously selected spool\n' +
+                    '"Abort": …to the new spool'
+                )
+            }
             // api-call
             var databaseId = -1
             if (spoolItem != null){
                 databaseId = spoolItem.databaseId();
+                var alreadyInTool = self.getSpoolItemSelectedTool(databaseId);
+                if (alreadyInTool !== null) {
+                    alert('This spool is already selected for tool ' + alreadyInTool + '!');
```

Comment:
Because otherwise the request would just be ignored and the user might be confused why it's not accepting the toolchange. Just a helpful hint for the user :)

Reply:
hmm..I removed that popup and continue with the selection.

Reply:
May i ask why? After all, having the same spool in two slots shouldn't be possible really? If you don't like it, it might be nice to make that an option to enable in the settings :)

Reply:
I just removed the alert, the spool is changed and removed from the old tool. So, it is not possible to have one spool assigned to two tools.
Less popups the user needs to react....do you still thing an alert is needed?

Reply:
I definitely think some kind of message is needed. The thing is: This basically shouldn't happen. If the user selects a spool that is already in a different tool, that is probably a mistake. Either the user has misclicked (in which case they would want it to be ignored) or the user is unaware that the spool is currently loaded in a different tool.

I was thinking about just hiding spools that are selected elsewhere from the select menu, but that might confuse the user as well, if they have forgotten/are unaware it's loaded in another tool. I felt like the most straightforward solution was, if the user tries to do it, to tell them "hey, you can't do that, and if you're wondering why, here's why."

Another option might be to just gray out/disable the options in the select button and display a small label next to them with the tool number the spool is currently loaded in.

Reply:
But then the user needs to deselect the old spool first and then he/she is able to use this spool for an other tool....that is a little bit annoying.
A popup that will disappear after 5-10 seconds to inform the user about re-selecting would be one compromise or add an option in the plugin-settings (e.g. [X] prevent selecting spool that where already assigned...default value: unchecked)
...hmmmm multiple solutions possible :-)
I like the "gray out" option and/or the plugin-settings "prevent selecting"

I will raise a ticket for that feature.

Reply:
Another option would just be use a confirm() dialog instead. "Hey, wanna really do that? It's loaded in tool x. If you click OK it will be removed from that one."